### PR TITLE
[GH-692] Auto-size takes the sort arrow into account

### LINKED ISSUE #692: Bug: Auto-size Columns with sort arrow
**Summary:**
Auto Size does not take the sort arrow into account in the column header.

**Steps to reproduce the behavior:**
1. Create a new Roadmap board.
2. The initial board should display a board with a Priority column header truncated to Priorit...
3. Double click the resize handle on the right side of the Priority column
4. Priority column is still trucated.

**Expected behavior:**
Column header should not be truncated, but display the full column header.

If you remove the sort, you will see it sized correctly without the sort arrow.

**Screenshots:**
Screen Shot 2021-07-08 at 9 57 20 AM

**Platform:**
 - Browser and version: [Chrome 91.0.4472.114 ,]
 - OS and version: [macOS 11.4]

**Additional context:**

### PR DESCRIPTION
**Pull request by kamre**
#### Summary
Proper computation of total horizontal padding for svg element.

#### Ticket Link
Fixes #692

@sbishel FYI

### CODE REVIEW

**Review by sbishel — CHANGES_REQUESTED** [review]
Just minor change for readability, also would you add a unit test for this function.

File: webapp/src/utils.ts [hunk 1]
```diff
@@ -91,20 +91,39 @@ class Utils {
...
-                const childResults = Utils.getFontAndPaddingFromChildren(element.children, myResults.padding)
-                if (childResults.fontDescriptor !== '') {
-                    myResults.fontDescriptor = childResults.fontDescriptor
-                    myResults.padding = childResults.padding
+                if (element.tagName === 'svg') {
+                    // clientWidth already includes padding
+                    myResults.padding += element.clientWidth
+                    myResults.padding += Utils.getHorizontalBorder(style)
+                    myResults.padding += Utils.getHorizontalMargin(style)
+                } else {
```
[the remark this answers is not in the document]
Comment (kamre) [reply webapp/src/utils.ts:99]: @sbishel svg elements have [this type](https://developer.mozilla.org/en-US/docs/Web/API/SVGAnimatedString) for className:
image
How do you propose to move such check inside `switch` statement? We can't compare it with some static object.

**Comment by kamre** [comment]
@sbishel I would like to add a unit test for `getFontAndPaddingFromCell` but not sure how to do this properly.

Unit test written with help of the testing library [uses JSDOM library internally](https://testing-library.com/docs/dom-testing-library/intro) for the execution of the test. And [JSDOM doesn't have any layout engine](https://github.com/jsdom/jsdom/issues/1322) so it can't compute properly all the style information ([like clientWidth](https://github.com/jsdom/jsdom/issues/2310)) that is used inside `getFontAndPaddingFromCell`.

So how can this function be tested in the unit test?

Can/should we somehow mock [getComputedStyle](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) for all the elements in the tree that we pass to `getFontAndPaddingFromCell`?

File: webapp/src/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbishel) [reply webapp/src/utils.ts:99]: I mean to move it above the switch statement -
```
Array.from(children).forEach((element) => {
    if (element.tagName === 'svg') {
        // clientWidth already includes padding
        myResults.padding += element.clientWidth
        myResults.padding += Utils.getHorizontalBorder(style)
        myResults.padding += Utils.getHorizontalMargin(style)
    } else {
       switch (element.className) {
       ...}
    }
}
```

[the remark this answers is not in the document]
Comment (kamre) [reply webapp/src/utils.ts]: Agree, this makes sense. Done.

**Review by sbishel — APPROVED** [review]
@kamre - Your right, I don't see an easy way to write tests for these. Thanks for making the other change and your submission.

**Review by harshilsharma63 — APPROVED** [review]

**Comment by harshilsharma63** [comment]
This led me to find another bug - the auto-size doesn't work in Firefox. Created #817 for the same.

**Comment by jespino** [comment]
I'll merge this after the tests passes.

**Pull request merged**